pkg/catalog/types.go

```diff
@@ -1,92 +1,112 @@
-// Package catalog implements the MeshCataloger interface, which forms the central component in OSM that transforms
+// Package catalog implements the MeshCatalog, which forms the central component in OSM that transforms
 // outputs from all other components (SMI policies, Kubernetes services, endpoints etc.) into configuration that is
 // consumed by the the proxy control plane component to program sidecar proxies.
 // Reference: https://github.com/openservicemesh/osm/blob/main/DESIGN.md#5-mesh-catalog
 package catalog
 
 import (
-	"github.com/openservicemesh/osm/pkg/certificate"
-	"github.com/openservicemesh/osm/pkg/compute"
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/apis/policy/v1alpha1"
 	"github.com/openservicemesh/osm/pkg/endpoint"
 	"github.com/openservicemesh/osm/pkg/identity"
+	"github.com/openservicemesh/osm/pkg/k8s"
 	"github.com/openservicemesh/osm/pkg/logger"
+	"github.com/openservicemesh/osm/pkg/models"
 	"github.com/openservicemesh/osm/pkg/service"
-	"github.com/openservicemesh/osm/pkg/trafficpolicy"
 )
 
 var (
 	log = logger.New("mesh-catalog")
 )
 
-// MeshCatalog is the struct for the service catalog
-type MeshCatalog struct {
-	compute.Interface
-	certManager *certificate.Manager
-}
+type trafficDirection string
 
-// MeshCataloger is the mechanism by which the Service Mesh controller discovers all Envoy proxies connected to the catalog.
-type MeshCataloger interface {
-	compute.Interface
+const (
+	inbound  trafficDirection = "inbound"
+	outbound trafficDirection = "outbound"
+)
 
-	// ListOutboundServicesForIdentity list the services the given service identity is allowed to initiate outbound connections to
-	ListOutboundServicesForIdentity(identity.ServiceIdentity) []service.MeshService
+// Interface is an interface to be implemented by components abstracting Kubernetes, and other compute/cluster providers
+type Interface interface {
+	k8s.PassthroughInterface
```

Comment:
Similarly, let's expand out the k8s.PassthroughInterface, and delete those interfaces (move them into where they are used).

Can be done later though, but maybe add a TODO